### PR TITLE
Feature/create anonymize customer without any orders

### PR DESCRIPTION
**Pull request by GalloisLuca**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?   | Yes

### CODE REVIEW

File: src/Processor/AdvancedActions/AnonymizeCustomersNotLoggedBeforeProcessor.php [hunk 1]
```diff
@@ -7,36 +7,37 @@
...
     {
-        /** @var \DateTime $before */
-        $before = $form->getData()['before_date'];
-        $beforeClean = new \DateTime($before->format('Y-m-d'));
-
         $shopUsers = $this->entityManager
             ->createQueryBuilder()
             ->select('su')
-            ->from(ShopUserInterface::class, 'su')
+            ->from(get_class($this->shopUserFactory->createNew()), 'su')
```
Comment (oallain) [inline src/Processor/AdvancedActions/AnonymizeCustomersNotLoggedBeforeProcessor.php]: Shouldn't we use the `sylius.model.shop_user.class` parameter?

File: src/Resources/translations/messages.en.yaml [hunk 2]
```diff
@@ -9,6 +9,6 @@ sylius:
                     gdpr_title: GDPR
                 advanced_actions:
                     title: Advanced actions
-                    anonymize_customer_not_logged_before:
-                        label: Anonymize customer not logged before
+                    anonymize_customers_not_logged_before:
+                        label: Anonymize customers not logged before
         execute: Execute
```
Comment (oallain) [inline src/Resources/translations/messages.en.yaml]: Can you add blank line at the end of files ?

**Review by oallain — DISMISSED** [review]

File: src/DependencyInjection/CompilerPass/RegisterAdvancedActionsFormDataProcessorsPass.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
```
Comment (oallain) [inline src/DependencyInjection/CompilerPass/RegisterAdvancedActionsFormDataProcessorsPass.php]: No 😉

File: src/Processor/AdvancedActions/AnonymizeCustomersWithoutAnyOrdersBeforeProcessor.php [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+        $this->anonymizerProcessor->anonymizeEntities($customers);
+    }
+
+    private function removeNoneEligibleCustomers(array &$customers): array
+    {
+        /** @var CustomerInterface $customer */
+        foreach ($customers as $key => $customer) {
+            foreach ($customer->getOrders() as $order) {
+                if ($order->getState() === 'cart') {
```
Comment (oallain) [inline src/Processor/AdvancedActions/AnonymizeCustomersWithoutAnyOrdersBeforeProcessor.php]:
```suggestion
                if ($order->getState() === OrderInterface::STATE_CART) {
```

File: src/SynoliaSyliusGDPRPlugin.php [hunk 5]
```diff
@@ -14,6 +14,8 @@ final class SynoliaSyliusGDPRPlugin extends Bundle
 {
     use SyliusPluginTrait;
 
+    public const VERSION = '1.1.0';
```
Comment (oallain) [inline src/SynoliaSyliusGDPRPlugin.php]: Can you remove it ?

**Review by oallain — CHANGES_REQUESTED** [review]

**Review by oallain — DISMISSED** [review]

File: src/Processor/AdvancedActions/AnonymizeCustomersNotLoggedBeforeProcessor.php [hunk 6]
```diff
@@ -7,36 +7,40 @@
...
 {
     /** @var EntityManagerInterface */
     private $entityManager;
 
     /** @var AnonymizerProcessor */
     private $anonymizerProcessor;
 
-    public function __construct(EntityManagerInterface $entityManager, AnonymizerProcessor $anonymizerProcessor)
+    /** @var ParameterBagInterface */
+    private $parameterBag;
```
Comment (TheGrimmChester) [inline src/Processor/AdvancedActions/AnonymizeCustomersNotLoggedBeforeProcessor.php]: You can type hint that

File: src/Processor/AdvancedActions/AnonymizeCustomersWithoutAnyOrdersBeforeProcessor.php [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+use Symfony\Component\Form\FormInterface;
+use Synolia\SyliusGDPRPlugin\Processor\AnonymizerProcessor;
+
+class AnonymizeCustomersWithoutAnyOrdersBeforeProcessor implements AdvancedActionsFormDataProcessorInterface
+{
+    /** @var EntityManagerInterface */
+    private $entityManager;
```
Comment (TheGrimmChester) [inline src/Processor/AdvancedActions/AnonymizeCustomersWithoutAnyOrdersBeforeProcessor.php]: You can type hint that and all the others

File: tests/PHPUnit/Processor/AdvancedActions/AnonymizeCustomerWithoutAnyOrdersBeforeProcessorTest.php [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+use Sylius\Component\Core\Model\OrderInterface;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\Form\Forms;
+use Synolia\SyliusGDPRPlugin\Form\Type\Actions\AnonymizeCustomersWithoutAnyOrdersBeforeType;
+use Synolia\SyliusGDPRPlugin\Processor\AdvancedActions\CompositeAdvancedActionsFormDataProcessor;
+
+class AnonymizeCustomerWithoutAnyOrdersBeforeProcessorTest extends KernelTestCase
+{
+    /** @var EntityManagerInterface */
+    private $manager;
```
Comment (TheGrimmChester) [inline tests/PHPUnit/Processor/AdvancedActions/AnonymizeCustomerWithoutAnyOrdersBeforeProcessorTest.php]: You can type hint that

**Review by TheGrimmChester — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 9]
```diff
@@ -5,3 +5,4 @@ composer.lock
 
 /tests/Application/
 /install/Application/*.local
+.idea/
```
Comment (oallain) [inline .gitignore]: Already on master, so you have a conflict, can you remove it?

File: install/Application/.gitignore [hunk 10]
```diff
@@ -0,0 +1,3 @@
+.env.*.local
```
Comment (oallain) [inline install/Application/.gitignore]: In .gitignore in root directory we have
`/tests/Application/`
so you need to remove this file.

**Review by oallain — DISMISSED** [review]

**Review by oallain — APPROVED** [review]

**Review by TheGrimmChester — APPROVED** [review]

**Comment by oallain** [comment]
Thanks @GalloisLuca and @tuyetrinhvo :rocket:

**Pull request merged**